File: 10_mouse_input_review_peer/program_files/cpp/4mousemove.cpp
#include "splashkit.h"

int main()
{
    // colour variables
    double hue = 0.8;
    double saturation = 0.8;
    double brightness = 0.8;

    // mouse movement variable
    vector_2d movement;

    open_window("Mouse Movement", 800, 600);

    while (!window_close_requested("Mouse Movement"))
    {
        // process user input events
        process_events();

        // store the mouse movement direction and distance to the movement variable
        movement = mouse_movement();

        // change screen colour 'hue' if left mouse button is held down
        if (mouse_down(LEFT_BUTTON))
        {
            // change hue of screen colour based on horizontal mouse movement
            hue += movement.x / screen_width();
        }
        // change screen colour 'saturation' if right mouse button is held down
        if (mouse_down(RIGHT_BUTTON))
        {
            // change saturation of screen colour based on vertical mouse movement
            saturation += movement.y / screen_height();
        }

        // display current screen colour
        clear_screen(hsb_color(hue, saturation, brightness));
        refresh_screen(60);
    }

    close_all_windows();

    return 0;
}

File: 10_mouse_input_review_peer/program_files/python/4mousemove.py
from splashkit import *

def main():

    hue = 0.8
    saturation = 0.8
    brightness = 0.8


    window = open_window("Mouse Movement", 800, 600)

    while not window_close_requested(window):

        process_events()

        # Get the mouse movement
        movement = mouse_movement()

        # Update hue and saturation based on mouse movement
        if mouse_down(MouseButton.left_button):
            # Adjust hue based on horizontal mouse movement
            hue = (hue + movement.x / window_width(window)) % 1.0

        if mouse_down(MouseButton.right_button):
            # Adjust saturation based on vertical mouse movement
            saturation = min(max(saturation + movement.y / window_height(window), 0.0), 1.0)

        # Create the color using HSB values
        color = hsb_color(hue, saturation, brightness)

        # Clear the screen with the new color
        clear_screen(color)


        refresh_window(window)


    close_window(window)

if __name__ == "__main__":
    main()